test/integration/PciManagerTest.php

```diff
@@ -0,0 +1,153 @@
+        $this->requestMock = $this->createMock(\Request::class);
+        $this->itemsService = \taoItems_models_classes_ItemsService::singleton();
+        $this->itemsService->setModel($this->ontologyMock);
+        $this->portableElementService = new PortableElementService();
+        $this->testService = \taoTests_models_classes_TestsService::singleton();
+        $this->importService = ImportService::singleton();
+        \common_ext_ExtensionsManager::singleton()->getExtensionById('taoQtiItem');
+        $this->itemsScannerServiceMock = $this->createMock(ItemsScannerService::class);
+    }
+
+    /**
+     * @test
+     * @expectedException \oat\taoQtiItem\model\portableElement\exception\PortableElementException
+     */
+    public function unregisterRequestWithoutParameterTest()
+    {
+        $this->requestMock
+            ->expects($this->once())
+            ->method('hasParameter')
+            ->with($this->equalTo('typeIdentifier'))
+            ->willReturn(false);
+
+        $pciManager = new PciManagerForTest($this->requestMock);
+        $pciManager->unregister();
+    }
+
+    private function createRequestMockWithTypeIdentifier()
+    {
+        $this->requestMock
+            ->expects($this->once())
+            ->method('hasParameter')
+            ->with($this->equalTo('typeIdentifier'))
+            ->willReturn(true);
+
+        $this->requestMock
+            ->method('getParameter')
+            ->with($this->equalTo('typeIdentifier'))
+            ->willReturn('likertScaleInteraction');
+    }
+
+    /**
+     * @test
+     */
+    public function unregisterTest()
+    {
+        $this->createRequestMockWithTypeIdentifier();
+        $pciModelMock = $this->createMock(PciModel::class);
+        $IMSPciModelMock = $this->createMock(IMSPciModel::class);
+        $pciDataObjectMock = $this->createMock(PciDataObject::class);
+        $portableElementModelMock = $this->createMock(PortableElementModel::class);
+        $portableElementRegistryMock = $this->createMock(PortableElementRegistry::class);
+        $pciDataObjectMock->method('getModel')->willReturn($portableElementModelMock);
+        $portableElementModelMock->method('getRegistry')->willReturn($portableElementRegistryMock);
+        $portableElementRegistryMock->method('delete');
+        $this->itemsScannerServiceMock->method('isPciUsedInItems')->willReturn(false);
+
+        $pciRegistryMock = $this->createMock(PciRegistry::class);
+        $pciRegistryMock
+            ->method('getLatestVersion')
+            ->with('likertScaleInteraction')
+            ->willReturn($pciDataObjectMock);
+
+        $pciModelMock->method('getRegistry')->willReturn($pciRegistryMock);
+
+        $pciModels = [
+            $pciModelMock, $IMSPciModelMock
+        ];
+
+        $pciManager = new PciManagerForTest($this->requestMock, $pciModels);
+        $pciManager->unregister();
+    }
+}
+
+
+class PciManagerForTest extends PciManager
+{
+    private $request;
```

Comment:
Please add a docblock with typehint

Reply:
`$request` should be made protected here, since the inherited class defines this as `protected $request;`